docs/developer/implementing-a-feature.md

```diff
@@ -0,0 +1,62 @@
+   receive early feedback, consider opening a draft PR and requesting reviews from the maintainers. Draft PRs are an
+   effective means to solicit feedback and ensure that major architectural changes align with the project's goals and
+   standards.
+7. **Add or update unit tests** All features **must** be accompanied by unit tests that verify the functionality. Make
+   sure to thoroughly test the different scenarios and edge cases related to the feature to ensure robustness and
+   reliability. Additionally, open the code coverage report to ensure that the code you added has sufficient test
+   coverage. For instructions on writing and running unit tests, refer to the [testing](testing.md#unit-test-guidelines)
+   documentation.
+8. **Manually verify your changes**: Refer to the [manual testing](testing.md#manual-testing) section of the testing
+   documentation for instructions on how to manually test your changes.
+9. **Update any relevant documentation**: Here are some guidelines for updating documentation:
+    - **Gateway API Feature**: If you are implementing a Gateway API feature, make sure to update
+      the [Gateway API Compatibility](../gateway-api-compatibility.md) documentation.
+    - **New Use Case:** If your feature introduces a new use case, add an example of how to use it in
+      the [examples](../../examples) directory. This example will help users understand how to leverage the new feature
+      in their own deployments.
+    - **Installation Changes**: If your feature involves changes to the installation process of NKG, update
+      the [installation](../installation.md) documentation.
+    - **Command-line Changes**: If your feature introduces or changes a command-line flag or subcommand, update
+      the [cli help](../cli-help.md) documentation.
+    - **Other Documentation Updates**: For any other changes that affect the behavior, usage, or configuration of NKG,
+      review the existing documentation and update it as necessary. Ensure that the documentation remains accurate and
+      up to date with the latest changes.
+10. **Lint code**: See the [run the linter](quickstart.md#run-the-linter) section of the quickstart guide for
+    instructions.
+11. **Open PR**: Open a PR targeting the `main` branch of
+    the [nginx-kubernetes-gateway](https://github.com/nginxinc/nginx-kubernetes-gateway/tree/main) repository. The
+    entire `nginx-kubernetes-gateway` group will be automatically requested for review. If you have a specific or
+    different reviewer in mind, you can request them as well. Refer to the [pull request](pull-request.md) documentation
+    for expectations and guidelines.
+12. **Obtain the necessary approvals**: Work with code reviewers to maintain the required number of approvals.
+13. **Squash and merge**: Squash your commits locally, or use the GitHub UI to squash and merge. Only one commit per PR
```

Comment:
all our commits include PR number in the message. Shall we explicitly require that?

Reply:
That's in the pull request guidelines doc. Would it fit better here?